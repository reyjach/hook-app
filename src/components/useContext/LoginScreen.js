import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {


  const { setUser } = useContext( UserContext );

  const handleClick = () => {
    setUser({
      id: 123,
      name: 'carlos',
      email: 'carlos23@gmail.com'
    })
  }

  return (
    <div>
        <h1> LoginScreen </h1>
        <hr/>

        <button 
          className="btn btn-primary"
          onClick={ handleClick }>
          Login
        </button>
    </div>
  )
}
