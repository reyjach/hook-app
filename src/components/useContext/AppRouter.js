import React from 'react';
import { BrowserRouter as Router, Routes, Route} from "react-router-dom";

import { NavBar } from './NavBar';

import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';


export const AppRouter = () => {
  return (
    <Router>
        
        <NavBar></NavBar>
        <div className="container">
            <Routes>
                <Route index path="/about" element={<AboutScreen />}/>

                <Route index path="/login" element={<LoginScreen />}/>

                <Route index path="/" element={<HomeScreen />} />

                <Route path="*" element={<HomeScreen />} />

            </Routes>
        </div>
    </Router>
  )
}
