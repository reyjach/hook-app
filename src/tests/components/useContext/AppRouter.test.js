import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme from 'enzyme';
import { mount } from "enzyme";
import { AppRouter } from '../../../components/useContext/AppRouter';
import { UserContext } from "../../../components/useContext/UserContext";

describe('Pruebas en el componente AppRouter', () => { 

    Enzyme.configure({ adapter: new Adapter() });

    const user = {
        id: 123,
        name: 'carlos',
        email: 'carlos23@gmail.com'
    }


    const wrapper = mount(
        <UserContext.Provider value={[
            user
        ]}>
           <AppRouter />
        </UserContext.Provider>
    );

    test('Debe de mostrarse correctamente', () => { 
        expect( wrapper ).toMatchSnapshot();
     })
 })