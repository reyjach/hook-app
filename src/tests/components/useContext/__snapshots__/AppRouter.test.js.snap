// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AppRouter Debe de mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        useContext
      </a>
      <button
        aria-controls="navbarNavAltMarkup"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarNavAltMarkup"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
         HomeScreen 
      </h1>
      <hr />
      <pre />
    </div>
  </div>,
]
`;
